egs/tedlium/s5_r2/local/rnnlm/tuning/run_lstm_tdnn.sh

```diff
@@ -2,29 +2,52 @@
 
 # Copyright 2012  Johns Hopkins University (author: Daniel Povey)  Tony Robinson
 #           2017  Hainan Xu
-#           2017  Ke Li
+#           2018  Ke Li
 
-# rnnlm/train_rnnlm.sh: best iteration (out of 10) was 8, linking it to final iteration.
-# rnnlm/train_rnnlm.sh: train/dev perplexity was 78.4 / 147.8.
-# Train objf: -1556.00 -5.43 -5.15 -5.00 -4.90 -4.82 -4.75 -4.69 -4.63 -4.58
-# Dev objf:   -11.92 -5.70 -5.29 -5.16 -5.08 -5.04 -5.02 -5.00 -5.00 -5.00
+# rnnlm/train_rnnlm.sh: best iteration (out of 9) was 8, linking it to final iteration.
+# rnnlm/train_rnnlm.sh: train/dev perplexity was 94.1 / 155.1.
+# Train objf: -6.24 -5.45 -5.12 -4.95 -4.84 -4.74 -4.66 -4.59 -4.52 -4.46
+# Dev objf:   -11.92 -5.80 -5.32 -5.17 -5.10 -5.07 -5.05 -5.05 -5.04 -5.06
+
+# 1-pass results 
+# %WER 8.3 | 1155 27500 | 92.7 4.9 2.4 1.0 8.3 68.8 | -0.019 | /export/a12/ywang/kaldi/egs/tedlium/s5_r2/exp/chain_cleaned/tdnn_lstm1i_adversarial1.0_interval4_epoches7_lin_to_5_sp_bi/decode_looped_test/score_10_0.0/ctm.filt.filt.sys
+
+# 4-gram rescoring
+# %WER 7.8 | 1155 27500 | 93.1 4.5 2.4 0.9 7.8 66.4 | -0.089 | /export/a12/ywang/kaldi/egs/tedlium/s5_r2/exp/chain_cleaned/tdnn_lstm1i_adversarial1.0_interval4_epoches7_lin_to_5_sp_bi/decode_looped_test_rescore/score_10_0.0/ctm.filt.filt.sys
+
+# RNNLM lattice rescoring
+# %WER 7.4 | 1155 27500 | 93.4 4.2 2.4 0.8 7.4 65.0 | -0.117 | exp/decode_test_rnnlm_tedlium/score_10_0.0/ctm.filt.filt.sys
+
+# RNNLM nbest rescoring
+# %WER 7.7 | 1155 27500 | 93.1 4.3 2.6 0.8 7.7 65.8 | -0.855 | exp/decode_test_rnnlm_tedlium_nbest//score_10_0.0/ctm.filt.filt.sys
 
 # Begin configuration section.
+cmd=run.pl
+decode_cmd=run.pl
 dir=exp/rnnlm_lstm_tdnn
-embedding_dim=800
-lstm_rpd=200
-lstm_nrpd=200
-stage=-10
+embedding_dim=1024
+lstm_rpd=256
+lstm_nrpd=256
+stage=0
 train_stage=-10
 epochs=20
 
-. ./cmd.sh
-. utils/parse_options.sh
-[ -z "$cmd" ] && cmd=$train_cmd
+# variables for lattice rescoring
+run_lat_rescore=true
+run_nbest_rescore=true
+decode_dir_suffix=rnnlm_tedlium
+ac_model_dir=/export/a12/ywang/kaldi/egs/tedlium/s5_r2/exp/chain_cleaned/tdnn_lstm1i_adversarial1.0_interval4_epoches7_lin_to_5_sp_bi
```

Comment:
There shouldn't be absolute pathnames in these scripts, it's supposed to work when someone runs the checked-in scripts.     Also, this tuning/ directory should contain numbered scripts with suffixes like _1a.sh, _1b.sh, etc, and there should be a link from one level above to here.

Reply:
sure, I'll look into this today.